share/functions/help.fish

```diff
@@ -127,17 +132,24 @@ function help --description 'Show help for the fish shell'
         # Go to the web. Only include one dot in the version string
         set -l version_string (echo $version| cut -d . -f 1,2)
         set page_url https://fishshell.com/docs/$version_string/$fish_help_page
+        # We don't need a trampoline for a remote URL.
+        set need_trampoline
     end
 
-    # OS X /usr/bin/open swallows fragments (anchors), so use osascript
-    # Eval is just a cheesy way of removing the hash escaping
-    if test "$fish_browser" = osascript
-        set -l opencmd 'open location "'(eval echo $page_url)'"'
-        osascript -e 'try' -e $opencmd -e 'on error' -e $opencmd -e 'end try'
-        return
+    if set -q need_trampoline[1]
```

Comment:
Could we use `psub` to get a temporary file that fish deletes for us for free, simplifying the code?